lib/ansible/plugins/strategy/__init__.py

```diff
@@ -75,7 +75,7 @@ def results_thread_main(strategy):
     while True:
         try:
             result = strategy._final_q.get()
-            if type(result) == object:
+            if type(result) is type(_sentinel):
```

Comment:
Ah, jimi-c is probably the only one that can tell us if that is a good change or not.  I realized when I was looking at it that this is crossing the worker fork boundary so I do not know that _sentinel that comes from _final_q is the same as the sentinel that is defined in the file.  (For other people looking at this, isinstance should definitely be avoided because isinstance follows inheritance and everything inherits from object).